### PR TITLE
feat: waitUntilFirstFrameRendered; perf: Player.dispose event loop clean-up; refactor: API improvement etc.

### PR DESCRIPTION
**Pull request by alexmercerind**
### package:media_kit

- fix: `Player.dispose` event loop clean-up
- refactor: `Player` implementation clean-up
- feat: `Initializer.dispose`
  - feat: `InitializerIsolate.dispose`
  - feat: `InitializerNativeEventLoop.dispose`
- feat: `PlatformPlayer.waitForVideoControllerInitializationIfAttached`

### package:media_kit_libs_android_video

- build: bump dependencies

### package:media_kit_libs_android_audio

- build: bump dependencies

### package:media_kit_native_event_loop

- feat: `MediaKitEventLoopHandler::Dispose`
- perf: switch to `std::condition_variable`(s) for synchronization

### package:media_kit_video

- feat: `VideoController.waitUntilFirstFrameRendered`
- refactor: clean-up package structure
- refactor: remove `VideoController.dispose`
- refactor: `VideoController.create` -> `VideoController` constructor
- fix(android): add `av1` to `hwdec-codecs`
- fix(android): use `--vo=gpu` + `--hwdec=mediacodec-copy` /w `enableHardwareAcceleration`

### CODE REVIEW

**Comment by alexmercerind** [comment]
I don't have any issues on Windows, GNU/Linux & Android.

**Comment by alexmercerind** [comment]
I also merged #128 progress internally; since it makes some notable changes to package structure. However, it's not "plugged".

File: media_kit_video/lib/src/video_controller/web_video_controller/real.dart [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      null,
+      true,
+    );
+    // Register [_dispose] for execution upon [Player.dispose].
+    player.platform?.release.add(controller._dispose);
+
+    // Retrieve the [html.VideoElement] instance from [js.context].
+    controller._element = js.context[_kInstances][handle];
+    // Register the [html.VideoElement] as platform view.
+    platformViewRegistry.registerViewFactory(
```
Comment (birros) [inline media_kit_video/lib/src/video_controller/web_video_controller/real.dart:53]: I have this error with `flutter 3.10.1-stable`, but does not prevent the build from succeeding on non-web platforms:

```shell
$ cd media_kit_video
$ flutter analyze
Analyzing media_kit_video...                                            

  error • Undefined name 'platformViewRegistry' • lib/src/video_controller/web_video_controller/real.dart:53:5 • undefined_identifier

1 issue found. (ran in 2.5s)
```

File: media_kit/lib/src/player/libmpv/core/initializer_isolate.dart [hunk 2]
```diff
@@ -162,13 +194,26 @@ abstract class InitializerIsolate {
     while (true) {
       completer = Completer();
       final event = mpv.mpv_wait_event(handle, _isReleaseMode ? -1 : 0.1);
```
Comment (birros) [inline media_kit/lib/src/player/libmpv/core/initializer_isolate.dart:196]: On macOS, issue #180 is still effective, is the infinite timeout really necessary in release mode?

See https://github.com/alexmercerind/media_kit/issues/180#issuecomment-1542082084

```suggestion
      final event = mpv.mpv_wait_event(handle, 0.1);
```

**Review by birros — APPROVED** [review]
LGTM!

Modulo my comment about issue #180.

File: media_kit_video/lib/src/video_controller/web_video_controller/real.dart (hunk 1 above)
Comment (alexmercerind) [reply media_kit_video/lib/src/video_controller/web_video_controller/real.dart:53]: Was this resolved for you?

Comment (alexmercerind) [reply media_kit_video/lib/src/video_controller/web_video_controller/real.dart:54]: It should be good now. One more conditional import.

File: media_kit/lib/src/player/libmpv/core/initializer_isolate.dart (hunk 2 above)
Comment (alexmercerind) [reply media_kit/lib/src/player/libmpv/core/initializer_isolate.dart:196]: I'm not sure about this one.

It degrades performance quite a bit. I don't want to look bad from performance stand-point. In debug mode, we enable it to keep "hot-reload" functional (& optimal performance isn't a goal in debug builds anyway).

Maybe we can skip posting `MPV_EVENT_NONE` outside the `Isolate`.

**Pull request merged**